src/app/lib/usePreventChangeOnNumberInputScroll.ts

```diff
@@ -0,0 +1,47 @@
+import { RefObject, useRef } from 'react'
+import { TextInputProps } from 'grommet/es6/components/TextInput'
+
+export const usePreventChangeOnNumberInputScroll = ({
+  onWheel,
+  onFocus,
+  onMouseOut,
+}: Pick<TextInputProps, 'onWheel' | 'onFocus' | 'onMouseOut'> = {}): Pick<
+  TextInputProps,
+  'onWheel' | 'onFocus' | 'onMouseOut' | 'ref'
+> => {
+  const inputRef = useRef<HTMLInputElement>(null) as RefObject<HTMLInputElement>
+  const isFocused = useRef(false)
+
+  return {
+    onWheel: e => {
+      onWheel?.(e)
+
+      const target = e.target as HTMLElement
+
+      // Prevents input value change
+      target.blur()
+      // Prevents scrolling
+      e.stopPropagation()
+
+      if (!isFocused.current) return
+
+      // Focus after blur
+      setTimeout(() => {
+        target.focus()
+      }, 0)
+    },
+    ref: inputRef,
+    onFocus: e => {
+      onFocus?.(e)
+
+      isFocused.current = true
+    },
+    onMouseOut: e => {
+      onMouseOut?.(e)
+
+      if (document.activeElement === inputRef?.current) {
+        isFocused.current = false
+      }
+    },
+  }
+}
```

Comment:
This implementation loses focus, quite often
![Peek 2024-05-31 23-27](https://github.com/oasisprotocol/oasis-wallet-web/assets/3758846/c2389bb3-59bd-4f93-a37d-1623881d9210)

Reply:
Fix and simplify with
```suggestion
export const usePreventChangeOnNumberInputScroll = (): Pick<TextInputProps, 'onWheel'> => {
  return {
    onWheel: e => {
      const target = e.target as HTMLElement
      if (target !== document.activeElement) return

      // Prevents input value change
      target.blur()
      // Focus after blur
      setTimeout(() => {
        target.focus()
      }, 0)
    },
  }
}
```

Reply:
Yeah, looked into detail and `onMouseOut` does not work perfectly. Initially I used `onBlur` event, which should be the optimal way to check if focus is lost. But since I was blurring the element inside `onWheel` to prevent value change, it did not work as expected.

Either way, your solution works. The only issue I have is an unit test, where after wheel event the `activeElement` becomes `body` elements. But seems to work fine in browser environment.

Thanks for taking the time, to look for better solution 👍